pkg/datapath/linux/node.go

```diff
@@ -431,23 +431,100 @@ func (n *linuxNodeHandler) NodeUpdate(oldNode, newNode node.Node) error {
 	return nil
 }
 
-func (n *linuxNodeHandler) enableIPsec(newNode *node.Node) {
+func upsertIPsecLog(err error, spec string, loc, rem *net.IPNet, spi uint8) {
+	scopedLog := log.WithFields(logrus.Fields{
```

Comment:
but then it has to be duplicated from multiple places.

Reply:
Apparently not. For context, Andre did some benchmarks of this kind of thing before and measured sizeable improvements to performance for hot code (https://github.com/cilium/cilium/pull/8273).

If it's a pain, I wouldn't worry about it since this particular case shouldn't be hit super often.